### PR TITLE
Update version of NYPL Reservoir to 3.1.1 

### PR DESCRIPTION
**Pull request by avertrees**
## Ticket:

- JIRA ticket [DR-2924](https://jira.nypl.org/browse/DR-2924)

## This PR does the following:

- Update version of NYPL Reservoir to 3.1.1 
- fix focus link in mobile nav (only breaking change of the update that I found so far)
- remove changelog update that mentions commenting out the feedback form.

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "winston": "3.9.0"
   },
   "devDependencies": {
-    "@nypl/design-system-react-components": "2.1.3",
+    "@nypl/design-system-react-components": "^3.0.0",
```
Comment (EdwinGuzman) [inline package.json:52]: Pin the version by removing the caret `^`.

File: src/components/navMenu/mobileNavMenu.tsx [hunk 2]
```diff
@@ -7,9 +7,9 @@ import {
   List,
   Box,
   HorizontalRule,
+  FocusLock,
```
Comment (EdwinGuzman) [inline src/components/navMenu/mobileNavMenu.tsx:10]: This still works well.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
There are two issues which I'm not sure if it was introduced by the new version or overrides in the app that are no longer and affecting the styling.

On desktop there should be more spacing below the text input: 
Screenshot 2024-04-19 at 5 03 28 PM

On mobile, the searchbar "Search" button should be on the same row 
Screenshot 2024-04-19 at 5 03 34 PM

@7emansell might know more about this because the header had a few custom media query styles. Let's take a look next week.

**Comment by EdwinGuzman** [comment]
The hero image is not loading... I think it may be related to the DS and I will take a closer look.

**Comment by EdwinGuzman** [comment]
IIIF images don't load in the Hero and that will be fixed in https://github.com/NYPL/nypl-design-system/pull/1582. This is a Chakra bug and a workaround will be added in the next release. When that's out, update this PR to `3.1.1` -- we'll let you know when it's published.

For now, you should continue to fix the visual discrepancies from this initial `3.0.0` update. And if you can review the DS PR, that'd be great.

**Comment by 7emansell** [comment]
Visual discrepancies in the header/search bar fixed– had to do with `Button` and `Checkbox` defaults

File: src/components/navMenu/navMenu.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ const NavMenu = ({ render }) => {
       aria-label={text}
       key={text}
       sx={{
-        mb: "0px",
+        marginBottom: "0px",
```
Comment (EdwinGuzman) [inline src/components/navMenu/navMenu.tsx:13]: Is this to be more verbose or was it an actual issue?

Comment (7emansell) [reply src/components/navMenu/navMenu.tsx:13]: to be more verbose/consistent across styling

**Comment by EdwinGuzman** [comment]
> I'm also seeing how it's a little off but I'm thinking I can let Lily to tell me what to adjust if she sees an issue? 

Yes.

**Comment by EdwinGuzman** [comment]
Update this to DS v3.1.1 and it's good to go!

**Review by 7emansell — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,9 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
-### Removed
-
-- temporarily remove feedback button (DR-2918)
+### Updated
+-  Update version of NYPL Reservoir to 3.0.0 (DR-2924)
```
Comment (EdwinGuzman) [inline CHANGELOG.md:11]: Update to 3.1.1.

**Review by EdwinGuzman — APPROVED** [review]
Looks good just minor changelog update.

**Comment by EdwinGuzman** [comment]
And should be merged after the release on Monday.

**Comment by croyfish** [comment]
@avertrees 
Do you mind merging latest `main` into this PR branch? I'm needing to rebase https://github.com/NYPL/digital-collections/pull/130/files into this one, but now that branch is ahead.

**Comment by avertrees** [comment]
@croyfish I merged the latest version of main and the fallback image branch into this branch, I aslo changed the base branch to be [release/0.1.8](https://github.com/NYPL/digital-collections/tree/release/0.1.8). 
I'm going to merge this branch into the release branch, do you mind changing the base branch of your PR to be [release/0.1.8](https://github.com/NYPL/digital-collections/tree/release/0.1.8)?

**Pull request merged**